nemo_aligner/models/nlp/gpt/gpt_steerlm_model.py

```diff
@@ -0,0 +1,357 @@
+                if parallel_state.is_pipeline_first_stage():
+                    required_keys.update(("tokens", "position_ids"))
+                if parallel_state.is_pipeline_last_stage():
+                    required_keys.update(("labels", "loss_mask", "weight", "avg_num_valid_tokens_in_ub"))
+            if self.get_attention_mask_from_fusion and "attention_mask" in required_keys:
+                required_keys.remove("attention_mask")
+            batch = {key: val.cuda(non_blocking=True) if key in required_keys else None for key, val in batch.items()}
+
+            # slice batch along sequence dimension for context parallelism
+            batch = self.get_batch_on_this_context_parallel_rank(batch)
+
+            # Model forward pass
+            forward_args = {
+                "input_ids": batch["tokens"],
+                "position_ids": batch["position_ids"],
+                "attention_mask": None if self.get_attention_mask_from_fusion else batch["attention_mask"],
+                "labels": batch["labels"],
+                "loss_mask": batch["loss_mask"],
+            }
+
+            if not self.mcore_gpt:
+                forward_args["checkpoint_activations_all_layers"] = checkpoint_activations_all_layers
+                if not self.use_loss_mask:
+                    forward_args.pop("loss_mask")
+            else:
+                # TODO: @eharper can we add this to mcore?
+                forward_args.pop("loss_mask")
+
+                if "cu_seqlens" in batch:  # packed sequence from GPTSFTPackedDataset
+                    # these args are passed eventually into TEDotProductAttention.forward()
+                    cu_seqlens = batch["cu_seqlens"].squeeze()  # remove batch size dimension (mbs=1)
+                    # remove -1 "paddings" added in collate_fn
+                    if cu_seqlens_argmin is not None:
+                        cu_seqlens = cu_seqlens[: cu_seqlens_argmin.item()]
+                    else:
+                        cu_seqlens = cu_seqlens[: torch.argmin(cu_seqlens)]
+
+                    try:
+                        from megatron.core.packed_seq_params import PackedSeqParams
+                    except (ImportError, ModuleNotFoundError) as e:
+                        mcore_version = packaging.version.Version(version("megatron-core"))
+                        logging.error(
+                            f"megatron-core v{mcore_version} does not support training with packed sequence. "
+                            "Please use megatron-core >= 0.5.0, or set model.data.train_ds.packed_sequence=False"
+                        )
+                        raise e
+
+                    forward_args["packed_seq_params"] = PackedSeqParams(
+                        cu_seqlens_q=cu_seqlens,
+                        cu_seqlens_kv=cu_seqlens,
+                        max_seqlen_q=max_seqlen,
+                        max_seqlen_kv=max_seqlen,
+                        qkv_format="thd",
+                    )
```

Comment:
```suggestion
```
aligner doesn't support non mcore models. and packed sequence length is not tested on steerlm2, so we should remove